File: frontend/src/types/assistant.ts
export type IntentType = 'DOCUMENT' | 'GITHUB' | 'JIRA' | 'GENERAL_PROJECT';

export interface AssistantSource {
  type: string;
  filename?: string;
  document_id?: number;
  chunk_id?: number;
  chunk_index?: number;
  repo?: string;
  issue_key?: string;
  project?: string;
  distance?: number;
}

export interface AssistantAskRequest {
  question: string;
  top_k?: number;
  document_id?: number | null;
}

export interface AssistantAskResponse {
  question: string;
  intent: IntentType;
  answer: string;
  sources: AssistantSource[];
  retrieved_count: number;
}

export interface ChatMessage {
  id?: number;
  role: 'user' | 'assistant';
  content: string;
  created_at?: string;
  intent?: IntentType;
  sources?: AssistantSource[];
}

export interface ConversationSession {
  id: number;
  title: string;
  created_at: string;
  updated_at: string;
  messages?: ChatMessage[];
}

export interface ConversationListResponse {
  total: number;
  page: number;
  per_page: number;
  items: ConversationSession[];
}

export interface ChatRequest {
  question: string;
  conversation_id?: number | null;
  top_k?: number;
  document_id?: number | null;
}

export interface ChatResponse {
  conversation_id: number;
  question: string;
  intent: IntentType;
  answer: string;
  sources: AssistantSource[];
  retrieved_count: number;
}

File: Backend/app/schemas/assistant.py
"""
Pydantic schemas for AI Project Assistant Orchestration API.
"""
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field, field_validator


class AssistantAskRequest(BaseModel):
    question: str = Field(
        ...,
        description="The question to ask the AI Project Assistant.",
        example="What does the documentation say about model evaluation?",
    )
    top_k: int = Field(
        default=5,
        ge=1,
        le=20,
        description="Maximum number of relevant document context chunks or items to retrieve.",
    )
    document_id: Optional[int] = Field(
        default=None,
        ge=1,
        description="Optional document ID to restrict search within a specific document.",
    )

    @field_validator("question")
    @classmethod
    def validate_question(cls, v: str) -> str:
        if not v or not v.strip():
            raise ValueError("Question cannot be empty or whitespace only.")
        v_clean = v.strip()
        if len(v_clean) > 2000:
            raise ValueError("Question exceeds maximum length of 2000 characters.")
        return v_clean


class AssistantSource(BaseModel):
    type: str = Field(..., description="Type of source: 'document', 'github', 'jira', or 'general'")
    filename: Optional[str] = Field(default=None, description="Filename for document sources")
    document_id: Optional[int] = Field(default=None, description="Document ID for document sources")
    chunk_id: Optional[int] = Field(default=None, description="Chunk ID for document sources")
    repository: Optional[str] = Field(default=None, description="Repository name for GitHub sources")
    project: Optional[str] = Field(default=None, description="Project key for Jira sources")
    details: Optional[Dict[str, Any]] = Field(default=None, description="Additional source metadata")


class AssistantAskResponse(BaseModel):
    question: str
    intent: str = Field(..., description="Determined intent: 'DOCUMENT', 'GITHUB', 'JIRA', or 'GENERAL_PROJECT'")
    answer: str
    sources: List[AssistantSource] = Field(default_factory=list)
    retrieved_count: int = Field(default=0)
